src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -430,6 +441,23 @@ protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbExc
                 new BufferedReader(new InputStreamReader(partitionInputStream)),
                 CSVFormat.newFormat(delimiter).withQuote(quote).withEscape(escape));
         iterator = parser.iterator();
+
+        try {
+          if (!iterator.hasNext()) {
+            flagAsIncomplete = true;
+          }
+        } catch (Exception e) {
+          /*
+           * FIX ME: If we hit an exception for a malformed row (in case of quotes for example), we mark this as the
+           * last row
+           */
+          if (e.getMessage() != null && e.getMessage().contains(truncatedQuoteErrorMessage)) {
+            onLastRow = true;
+          } else {
+            throw e;
+          }
+        }
+
         for (int i = 0; i < numberOfSkippedLines; i++) {
```

Comment:
We can still blow up here if a header line after the first line is truncated within a quote, right? Also, what happens if not all headers are contained within the first partition? And shouldn't the header skipping be conditioned on being the first worker (we'd need to handle the case where header lines begin in subsequent partitions, though)?